### ISSUE TITLE
`Trace[IO]`

### ISSUE THREAD

**Issue by kubukoz** [tpolecat/natchez#296]
Starting with https://github.com/typelevel/cats-effect/pull/1393, cats-effect (just like other effect systems) has a way to propagate fiber-local state without `ReaderT` using `FiberLocal[F, A]`.

Additionally, to encapsulate the common pattern of having essentially `ReaderT[F, Ref ...]`, @raasahsan has opened https://github.com/typelevel/cats-effect/pull/1822.

We should investigate how/if these could be used to get, say, `IO[Trace[IO]]`.

**Comment by tpolecat** [comment]
Yeah we should be able to go from EntryPoint to Trace[IO]. I'll hack that up.
